client/main.lua

```diff
@@ -0,0 +1,246 @@
+local PersistentNotiMap = {}
+local nuiReady
+
+-- Debug Print Notification
+local function DebugPrintInfo(style, duration, title, message, image, sound, custom, position, persistent)
+  if cfg.debugMode then
+    print('Notification | Style: ' .. tostring(style) .. '\n | Title: ' .. tostring(title) .. '\n | Message: ' .. tostring(message) .. '\n | Image URL: ' .. tostring(image) ..'\n | Duration: ' ..tostring(duration) .. '\n | Sound: ' .. tostring(sound) .. '\n | Custom: ' .. tostring(custom) .. '\n | Position: ' .. tostring(position) .. '\n | Persistent: ' .. tostring(persistent))
+  end
+end
+
+-- General Debug Print
+local function DebugPrint(msg)
+  if cfg.debugMode then
+    print(msg)
+  end
+end
+
+local function printError(msg)
+  local errMsg = ('^1[T-Notify Error] %s'):format(msg)
+  print(errMsg)
+end
+
+local function verifyTypes(notiTable, isPersistent)
+  -- Might add more type validation here idk
+  if not nuiReady then
+    printError('NUI Frame is not ready yet')
+    return false
+  end
+
+  if not notiTable.style or type(notiTable.style) ~= 'string' then
+    printError('Style cannot be nil or must be a string for notifications')
+    return false
+  end
+
+  if not isPersistent and (notiTable.duration and type(notiTable.duration) ~= 'number') then
+    printError('Duration has to be a number for notifications')
+    return false
+  end
+
+  if notiTable.sound and (type(notiTable.sound) ~= 'boolean' and type(notiTable.sound) ~= 'table') then
+    printError('Sound property must be either a boolean or table for notifications')
+    return false
+  end
+
+  if notiTable.position and type(notiTable.position) ~= 'string' then
+    printError('Position property must be a string for this notifications')
+    return false
+  end
+
+  if notiTable.image and type(notiTable.image) ~= 'string' then
+    printError('The image property must be a string for this notifications')
+    return false
+  end
+
+  if notiTable.icon and type(notiTable.icon) ~= 'string' then
+    printError('The icon property must be a string for this notifications')
+    return false
+  end
+
+  return true
+end
+
+--Triggers a notification in the NUI using supplied params
+local function SendNotification(style, duration, title, message, image, sound, custom, position, icon)
```

Comment:
Ideally, this function would be refactored to take in an arguments object instead of this mass of parameters

Reply:
I'll try refactoring the typesCheck along with this function.